### ISSUE TITLE
Exclude should ignore groups

### ISSUE THREAD

**Issue by WetRobot** [r-lib/lintr#2831]
This works

```r
lintr::lint(
  text = "x = 1",
  linters = lintr::assignment_linter(),
  exclude = "(?:a)|(?:b)"
)
```

But this

```r
lintr::lint(
  text = "x = 1",
  linters = lintr::assignment_linter(),
  exclude = "(a)|(b)"
)
```

raises an error that I think most users will not understand:

```
Error in `[.data.frame`(re_matches(lines, exclude, locations = TRUE),  : 
  undefined columns selected
```

**Comment by AshesITR** [comment]
Thanks for raising this. When fixing, we could also apply regex validation to other settings (nolint directives, for example).
